### PR TITLE
Executing Runtime API Calls from the Browser

### PR DESCRIPTION
**Pull request by tadeohepperle**
We want Runtime APIs to be executable in the Browser. This PR adds call execution for methods that do not need input values. 

Open Question: How far should we go in a direction similar to Polkadot.js providing input forms generated from type information, that can be filled out to submit storage requests and runtime api calls?

I guess one blocker for this right now is that it is not clear what format that input should be in, especially for e.g. `[u8;...]` values. We would like to have a way to convert hex or ss58 formatted binary data into scale values which are then converted into scale encoded blobs.

image

### CODE REVIEW

File: codegen/Cargo.toml [hunk 1]
```diff
@@ -3,52 +3,32 @@ name = "subxt_example_codegen"
...
 
-web = [
-    "subxt/web",
-    "js-sys",
-    "wasm-bindgen",
-    "wasm-bindgen-futures",
-    "serde-wasm-bindgen",
-    "console_error_panic_hook",
-]
+web = ["subxt/web", "js-sys", "wasm-bindgen", "wasm-bindgen-futures", "serde-wasm-bindgen", "console_error_panic_hook"]
```
Comment (jsdw) [inline codegen/Cargo.toml:11]: Separate lines is nicer imo :)

File: codegen/Cargo.toml [hunk 2]
```diff
@@ -3,52 +3,32 @@ name = "subxt_example_codegen"
...
 
-subxt = { version = "0.32.1", default-features = false, features = [
-    "jsonrpsee",
-    "unstable-light-client",
-], target_arch = "wasm32" }
+subxt = { version = "0.32.1", default-features = false, features = ["jsonrpsee", "unstable-light-client"], target_arch = "wasm32" }
 subxt-metadata = { version = "0.32.1" }
-subxt-codegen = { version = "0.32.1", default-features = false }
-subxt-signer = { version = "0.32.1", features = ["subxt"] }
+subxt-codegen = { git = "https://github.com/paritytech/subxt", branch = "master", default-features = false }
```
Comment (jsdw) [inline codegen/Cargo.toml:17]: Does this still need to be a git reference?

File: codegen/src/lib.rs [hunk 3]
```diff
@@ -425,7 +439,10 @@ impl<'a> ExampleGenerator<'a> {
             .enumerate()
             .map(|(i, type_id)| {
                 let variable_name = format!("key_{i}");
-                let type_path = type_gen.resolve_type_path(type_id).prune();
+                let type_path = type_gen
+                    .resolve_type_path(type_id)
+                    .expect("field not found")
```
Comment (jsdw) [inline codegen/src/lib.rs:444]: Can faulty user-provided metadata lead to this panic?

**Review by jsdw — APPROVED** [review]
Looks great to me so long as the UI works (I didn't try it!); just a few small comments :)

> Open Question: How far should we go in a direction similar to Polkadot.js providing input forms generated from type information, that can be filled out to submit storage requests and runtime api calls?

Just to write what I said aloud at some point; I wouldn't go too far this way right now; there's much less value in overlapping more with PJS UI and the more we add here, the harder it becomes to maintain it all. So for now I'd keep it focused, and down the line we might re-evaluate!

File: codegen/src/lib.rs (hunk 3 above)
Comment (tadeohepperle) [reply codegen/src/lib.rs]: That is a good points, I changed it now to return an error instead. I am sure there are also other places, e.g. in the subxt CLI tool or codegen, where we panic if some type ids are not valid.

**Pull request merged**